airbyte-cdk/python/airbyte_cdk/sources/declarative/schema/json_file_schema_loader.py

```diff
@@ -63,7 +68,8 @@ def get_json_schema(self) -> Mapping[str, Any]:
             raw_schema = json.loads(raw_json_file)
         except ValueError as err:
             raise RuntimeError(f"Invalid JSON file format for file {json_schema_path}") from err
-        return raw_schema
+        self.__resource = resource
+        return self._resolve_schema_references(raw_schema)
```

Comment:
Given that we're mostly copying these two methods from the existing code in `schema_helpers.py`, it might be interesting to make the JsonFileSchemaLoader a subclass of `ResourceSchemaLoader`. And we can refactor the `SchemaLoader` interface to name `get_json_schema()` to `get_schema()`. And this class `json_file_schema_loader` would just be responsible for the overriding `get_schema()` and can reuse the dependent methods that were originally duplicated.

It does expand the scope of changes a little bit, but would help us cut down on the code duplication. Does this all make sense?